addons/better-terrain/BetterTerrain.gd

```diff
@@ -360,7 +374,10 @@ func add_terrain(ts: TileSet, name: String, color: Color, type: int, categories:
 		if c < 0 or c >= ts_meta.terrains.size() or ts_meta.terrains[c][2] != TerrainType.CATEGORY:
 			return false
 	
-	ts_meta.terrains.push_back([name, color, type, categories])
+	if icon and not (icon.has("path") or (icon.has("source_id") and icon.has("coord"))):
```

Comment:
Don't need to check 'icon' because it's typed as `Dictionary` and defaulted. It's always going to be a valid dictionary.

Reply:
Oh wait, that's just checking if it's non-empty. Never mind!

Reply:
I could make it more explicit with `not icon.is_empty()` if that would be better